### PR TITLE
Unlimited programs should be opened for users

### PR DESCRIPTION
**Pull request by dchubar-fj**
**Problem/Motivation**
Classes with an unlimited number of spots are displayed as completely filled in Activity Finder with the Daxko back-end. The root of the problem is that Open Y doesn't check the `limited` flag in the Daxko response. Daxko back-end relies on the `available` field in the response:
```
         "availability":{
            "limited":true,
            "full":false,
            "max":20,
            "taken":0,
            "available":20,
            "has_waiting_list":false
         },
```
But unlimited classes don't have the `available` field:
```
         "availability":{
            "limited":false,
            "full":false,
            "taken":0,
            "has_waiting_list":true,
            "waiting_list_count":0,
            "show_waiting_list_count":false
         },
```

As result, unlimited classes are not available for registration in the Activity Finder, because the Register button is disabled
Screenshot from 2021-07-08 18-12-46
But they are open for registration in Daxko
Screenshot from 2021-07-08 18-14-16

## Steps for review

- [ ] Find a class with unlimited spots in the Activity Finder with the Daxko back-end
- [ ] Check program details
- [ ] Verify the register button is not disabled

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer. See [Code of Conduct](https://github.com/ymcatwincities/openy/wiki/Open-Y-Code-of-Conduct-and-Best-Practices).
- [ ] Try to use Conventional commit messages accordingly to the standard https://www.conventionalcommits.org/en/v1.0.0/#specification
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/9.x-2.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/9.x-2.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/9.x-2.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

File: modules/custom/openy_daxko2/src/OpenyActivityFinderDaxkoBackend.php [hunk 1]
```diff
@@ -714,6 +714,13 @@ public function getProgramsMoreInfo($request) {
       }
       $availability_note = $offeringResponse['details'][0]['registration_summaries'][0]['description'];
       $spots_available = $offeringResponse['details'][0]['availability']['available'];
+      $limited = $offeringResponse['details'][0]['availability']['limited'];
+
+      // Daxko doesn't provide a number of available spots for unlimited offers.
+      // But they should be open for users.
+      if ($availability_status === 'open' && !$spots_available && !$limited) {
+        $spots_available = 100;
```
Comment (podarok) [inline modules/custom/openy_daxko2/src/OpenyActivityFinderDaxkoBackend.php]: let's keep this as a constant of the current class, please

**Review by podarok — CHANGES_REQUESTED** [review]

**Comment by podarok** [comment]
@dchubar-fj could you put your description in module's readme.md?
Thank you

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build3277/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/3277/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build3277/

**Review by podarok — APPROVED** [review]
Code looks good
Let's improve readme a bit @dchubar-fj

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build3278/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/3278/

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build3278/

**Pull request merged**